packages/amplify-codegen-e2e-tests/src/__tests__/pull-codegen.test.ts

```diff
@@ -0,0 +1,131 @@
+import { 
+  createNewProjectDir, 
+  deleteProjectDir,
+  deleteProject,
+  initJSProjectWithProfile,
+  addApiWithSchemaAndConflictDetection,
+  amplifyPush,
+  amplifyPull,
+  getAppId,
+  DEFAULT_JS_CONFIG,
+  DEFAULT_ANDROID_CONFIG,
+  DEFAULT_IOS_CONFIG,
+  DEFAULT_FLUTTER_CONFIG,
+  AmplifyFrontendConfig,
+  getAdminApp,
+  amplifyPullSandbox,
+  getProjectSchema
+} from "amplify-codegen-e2e-core";
+import { existsSync } from "fs";
+import path from 'path';
+import { isNotEmptyDir, generateSourceCode } from '../utils';
+import { JSONUtilities } from 'amplify-cli-core';
+import { SandboxApp } from '../types/SandboxApp';
+
+
+
+const schema = 'simple_model.graphql';
+const envName = 'pulltest';
+const frontendConfigs: AmplifyFrontendConfig[] = [
+  DEFAULT_JS_CONFIG,
+  DEFAULT_ANDROID_CONFIG,
+  DEFAULT_IOS_CONFIG,
+  DEFAULT_FLUTTER_CONFIG
+];
+
+describe('Amplify pull in amplify app with codegen tests', () => {
+  let projectRoot: string;
+  let appId: string;
+  beforeAll(async () => {
+    projectRoot = await createNewProjectDir('pullCodegen');
+    await initJSProjectWithProfile(projectRoot, { envName, disableAmplifyAppCreation: false });
+    await addApiWithSchemaAndConflictDetection(projectRoot, schema);
+    await amplifyPush(projectRoot);
+    appId = getAppId(projectRoot);
+  });
+
+  afterAll(async () => {
+    const metaFilePath = path.join(projectRoot, 'amplify', '#current-cloud-backend', 'amplify-meta.json');
+    if (existsSync(metaFilePath)) {
+      await deleteProject(projectRoot);
+    }
+    deleteProjectDir(projectRoot);
+  });
+
+  it('should generate models and do not delete user src files by amplify pull in original project', async () => {
```

Comment:
We could add in description that this test if for JS project.
Also, we could use the `DEFAULT_JS_CONFIG` instead of `src` and `src/models` below.